tla-types/src/main/scala/at/forsyte/apalache/tla/typecheck/etc/ToEtcExpr.scala

```diff
@@ -0,0 +1,858 @@
+package at.forsyte.apalache.tla.typecheck.etc
+
+import at.forsyte.apalache.tla.lir.{ValEx, _}
+import at.forsyte.apalache.tla.lir.oper._
+import at.forsyte.apalache.tla.lir.values._
+import at.forsyte.apalache.tla.typecheck._
+import at.forsyte.apalache.tla.typecheck.parser.{DefaultType1Parser, Type1ParseError}
+
+/**
+  * ToEtcExpr takes a TLA+ expression and produces an EtcExpr.
+  * The most interesting part of this translation is dealing with the built-in operators.
+  * This translator is an extension of the ideas that appear in SignatureGenerator by Jure Kukovec.
+  *
+  * @see at.forsyte.apalache.tla.types.SignatureGenerator
+  *
+  * @author Igor Konnov
+  */
+class ToEtcExpr(varPool: TypeVarPool) extends EtcBuilder {
+  private val type1Parser = DefaultType1Parser
+
+  /**
+    * The counter that we use to produce fresh variables
+    */
+  private var nextVarNum = 0
+
+  /**
+    * Get the number just above the maximum variable number
+    * @return an upper bound on the variable numbers
+    */
+  def varNumUpperBound: Int = nextVarNum
```

Comment:
Same here. This must be the leftover after the old version, which can be safely removed.